### ISSUE TITLE
'pip3.13 install pillow' gives KeyError

### ISSUE THREAD

**Issue by johannesnoordanus** [python-pillow/Pillow#8089]
### What did you do?
```pip3.13 install pillow```

### What did you expect to happen?
Pil(low) installed. 

### What actually happened?
```pytb
 Traceback (most recent call last):
    File "/usr/local/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 354, in <module>
      main()
      ~~~~^^
    File "/usr/local/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 336, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
    File "/usr/local/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 119, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-u82hn5xj/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
             ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/tmp/pip-build-env-u82hn5xj/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
      self.run_setup()
      ~~~~~~~~~~~~~~^^
    File "/tmp/pip-build-env-u82hn5xj/overlay/lib/python3.13/site-packages/setuptools/build_meta.py", line 311, in run_setup
      exec(code, locals())
      ~~~~^^^^^^^^^^^^^^^^
    File "<string>", line 33, in <module>
    File "<string>", line 27, in get_version
  KeyError: '__version__'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /usr/local/bin/python3 /usr/local/lib/python3.13/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py get_requires_for_build_wheel /tmp/tmp00ch5v33
  cwd: /home/johannes/Downloads/Pillow-10.3 (1).0
  Getting requirements to build wheel ... error
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
```
### What are your OS, Python and Pillow versions?

* OS: Armbian (Ubuntu):
```
uname -a
Linux orangepi3b 6.8.10-edge-rockchip64 #1 SMP PREEMPT Fri May 17 10:15:12 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux
```
* Python: Python-3.13.0b1
* Pillow: 10.3

**Comment by aclark4life** [comment]
@johannesnoordanus Try installing from the main development branch maybe, we [don't support 3.13 yet](https://pillow.readthedocs.io/en/stable/installation/python-support.html).

**Comment by radarhere** [comment]
Pillow should work with 3.13 alpha 6. The change that caused this problem only appeared in 3.13 beta 1. It is fixed by #8050. On July 1, Pillow 10.4.0 is scheduled to be released, which should include that, and also include Python 3.13 beta wheels.

However, Pillow will not officially support Python 3.13 until Pillow 11.0.0 on October 15, because that will be the first release after Python 3.13 final is released.

**Comment by radarhere** [comment]
Closing as a duplicate of #8075
